lib/internal/modules/esm/public_loader_proxy.js

```diff
@@ -0,0 +1,42 @@
+const { Worker } = require('internal/worker');
+
+
+/**
+ * The communication vehical constituting all memory shared between the main and the worker
+ */
+const comsChannel = new SharedArrayBuffer(2048);
+/**
+ * The lock/unlock segment of the shared memory. Atomics require this to be a Int32Array. This
+ * segment is used to tell the main to sleep when the worker is worker, and vice verse (for the
+ * worker to sleep whilst the main is working).
+ */
+const lock = new Int32Array(comsChannel, 0, 4);
+/**
+ * The request & response segment of the shared memory. TextEncoder/Decoder (needed to convert
+ * requests & responses into a format supported by the coms channel) reads and writes with
```

Comment:
```suggestion
 * requests & responses into a format supported by the comms channel) reads and writes with
```

Why is the other `Int32Array` and this is `Uint8Array`?

Reply:
The comments tell you why 🙂

Reply:
I saw the comment explaining why `Int32Array`, but why `Uint8Array`?

Reply:
It's lines 15–17 of this file, right above this PR thread (the mobile app won't let me copy+paste). It starts with `TextEncoder/Decoder` (the reason is because that's the numerical datatype TextEncoder/Decoder can handle).

Maybe it doesn't need to be numerically encoded at all? 🤔 it goes into a SharedArrayBuffer, and the first slice of that is used by the Atomic, but the rest of it isn't. It's probably smaller, but I didn't check. TextEncoder/Decoder + Uint8Array was Guy's suggestion—I can ask him.